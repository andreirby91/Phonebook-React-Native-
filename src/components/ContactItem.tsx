import React from 'react';
import {View, Text, StyleSheet, Animated, TouchableOpacity} from 'react-native';
import Swipable from 'react-native-gesture-handler/Swipeable';
import FontistoIcon from 'react-native-vector-icons/Fontisto';

type Props = {
  name: string;
  phone: string;
  sex: string;
  country: string;
  code: string;
  onRightPress: () => void;
};

type RightActions = {
  progress: any;
  dragX: any;
  onPress: () => void;
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  actionText: {
    color: '#fff',
    fontWeight: '600',
    padding: 18,
    fontSize: 18,
  },
  rightAction: {
    backgroundColor: '#dd2c00',
    justifyContent: 'center',
    alignItems: 'flex-end',
    borderRadius: 4,
  },
  text: {
    fontSize: 18,
  },
  icon: {
    fontSize: 35,
    borderWidth: 0.5,
    borderRadius: 5,
    padding: 5,
    borderColor: 'grey',
    marginRight: 10,
  },
});

const ContactItem: React.FC<Props> = ({
  name,
  phone,
  sex,
  country,
  code,
  onRightPress,
}) => {
  const RightActions = ({progress, dragX, onPress}: RightActions) => {
    const scale = dragX.interpolate({
      inputRange: [-100, 0],
      outputRange: [1, 0],
      extrapolate: 'clamp',
    });

    return (
      <TouchableOpacity onPress={onPress}>
        <View style={styles.rightAction}>
          <Animated.Text style={[styles.actionText, {transform: [{scale}]}]}>
            Delete
          </Animated.Text>
        </View>
      </TouchableOpacity>
    );
  };

  return (
    <Swipable
      renderRightActions={(progress, dragX) => (
        <RightActions
          progress={progress}
          dragX={dragX}
          onPress={onRightPress}
        />
      )}>
      <View style={styles.container}>
        {sex === 'male' ? (
          <FontistoIcon name="male" style={styles.icon} />
        ) : (
          <FontistoIcon name="female" style={styles.icon} />
        )}

        <View>
          <Text numberOfLines={1} style={[styles.text, {marginBottom: 5}]}>
            {name}
          </Text>
          <Text
            numberOfLines={1}
            style={[
              styles.text,
              {marginBottom: 5},
            ]}>{`${code} ${phone} (${country})`}</Text>
        </View>
      </View>
    </Swipable>
  );
};

export default ContactItem;
