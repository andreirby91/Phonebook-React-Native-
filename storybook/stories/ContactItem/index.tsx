import React from 'react';
import ContactItem from '../../../src/components/ContactItem';

const ContactItemStory = () => {
  return <ContactItem />;
};

export default ContactItemStory;
