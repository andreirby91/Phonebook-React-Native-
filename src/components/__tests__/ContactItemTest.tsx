import React from 'react';
import renderer from 'react-test-renderer';

import ContactItem from '../ContactItem';

it('renders correctly with defaults', () => {
  const name = 'Andrei Beraru';
  const sex = 'Male';
  const phone = '0763111111';
  const country = 'Romania';
  const code = '40';

  const button = renderer
    .create(
      <ContactItem
        name={name}
        sex={sex}
        phone={phone}
        country={country}
        code={code}
        onRightPress={() => {}}
      />,
    )
    .toJSON();
  expect(button).toMatchSnapshot();
});
