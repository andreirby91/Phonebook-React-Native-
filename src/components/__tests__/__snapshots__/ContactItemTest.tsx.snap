// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
        "flexDirection": "row-reverse",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "backgroundColor": "#dd2c00",
            "borderRadius": 4,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "600",
              "padding": 18,
              "transform": Array [
                Object {
                  "scale": 0,
                },
              ],
            }
          }
        >
          Delete
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontFamily": "Fontisto",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î¦¤
      </Text>
      <View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "marginBottom": 5,
              },
            ]
          }
        >
          Andrei Beraru
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 18,
              },
              Object {
                "marginBottom": 5,
              },
            ]
          }
        >
          40 0763111111 (Romania)
        </Text>
      </View>
    </View>
  </View>
</View>
`;
